pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -82,6 +82,8 @@ type AuthInfo struct {
 	ClientKeyData []byte `json:"client-key-data,omitempty"`
 	// Token is the bearer token for authentication to the kubernetes cluster.
 	Token string `json:"token,omitempty"`
+	// ActAs is the username to act-as.
+	ActAs string `json:"act-as,omitempty"`
```

Comment:
> Seems unlikely we'd actually want this set in a config file?

I wasn't sure.  One significant use case is an `su` style interaction where you have an admin who doesn't normally have the power to perform certain actions, but can escalate himself when he needs to.  For one offs, he'd use `--as=`, but why shouldn't he be allowed to create a profile with it if he wants?  I wasn't planning to make it easy, just possible (and it is today using `kubectl config set`.

If you'd prefer, I can pull it out of the serialized format, but I'd like to keep it on the internal format to allow cleaner object merging.